simpleclient_servlet/src/main/java/io/prometheus/client/filter/MetricsFilter.java

```diff
@@ -0,0 +1,98 @@
+package io.prometheus.client.filter;
+
+import io.prometheus.client.Histogram;
+import org.apache.commons.lang3.StringUtils;
+
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import java.io.IOException;
+
+/**
+ * The MetricsFilter class exists to provide a high-level filter that enables tunable collection of metrics for Servlet
+ * performance.
+ *
+ * By default, this filter will provide metrics that distinguish 3 levels deep for the request path
+ * (including servlet context path), but can be configured with the {@code path-components} init parameter.
+ *
+ * The Histogram buckets can be configured with a {@code buckets} init parameter whose value is a comma-separated list of valid {@code double} values.
+ *
+ * The Histogram name itself can also be configured with a {@code metric-name} init parameter, whose value will be used directly.
+ */
+public class MetricsFilter implements Filter {
+    public static final String PATH_COMPONENT_PARAM = "path-components";
+    public static final String METRIC_NAME_PARAM = "metric-name";
+    public static final String BUCKET_CONFIG_PARAM = "buckets";
+    public static final String DEFAULT_FILTER_NAME = "servlet_request_latency_seconds";
```

Comment:
It's incorrect to have a default here, as if there's multiple uses that'll throw and exception and semantically each use would mean something different.